app/src/main/java/com/libs/facerecognition/NeuralModelProvider.java

```diff
@@ -0,0 +1,37 @@
+package com.libs.facerecognition;
+
+import android.content.Context;
+
+import com.R;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NeuralModelProvider {
+    private final static Map<String, NeuralModel> neuralModelsStorage = new HashMap<>();
+
+    /**
+     * Singleton NeuralModel instance getter. Initializes NeuralModel per name instance if not initialized earlier.
+     * Returns static NeuralModel instance.
+     *
+     * @param context       - app/activity context
+     * @param modelTag      - tag of the neural network model
+     * @param modelFilename - filename of the neural network model
+     * @return instance - singleton FaceNetModel instance
+     */
+    public static NeuralModel getInstance(Context context, String modelTag, String modelFilename) {
+        NeuralModel neuralModel = neuralModelsStorage.get(modelTag);
+

```

Comment:
maybe try to calculate model tag, cut ending of model name. It is 2 strings per model instead of one.